meta/data.go

```diff
@@ -662,6 +662,26 @@ func (di DatabaseInfo) RetentionPolicy(name string) *RetentionPolicyInfo {
 	return nil
 }
 
+// ShardInfos returns a list of all shards' info for the database.
+func (di DatabaseInfo) ShardInfos() []ShardInfo {
```

Comment:
This looks very like what @benbjohnson coded for `SHOW SHARD`. You will want to ignore shards in expired shard groups.